terraform/transform_orphan_count.go

```diff
@@ -50,20 +50,47 @@ func (t *OrphanResourceCountTransformer) Transform(g *Graph) error {
 }
 
 func (t *OrphanResourceCountTransformer) transformForEach(haveKeys map[addrs.InstanceKey]struct{}, g *Graph) error {
+	// If there is a no-key node, add this to the graph first,
+	// so that we can create edges to it in subsequent (StringKey) nodes.
+	// This is because the last item determines the resource mode for the whole resource,
+	// so if this (non-deterministically) happens to end up as the last one,
+	// that will change the resource's EachMode and our addressing for our instances
+	// will not work as expected
```

Comment:
In other words, we need to evaluate the orphaned resource before the in-memory state is updated to deal with the new for_each resource. I'm not sure if that makes anything more clear or not for the comment, so I'll let you decide if you want to change the wording or not.

Reply:
I've made a couple tweaks -- mind taking a look? I'm being a bit wordy to also try and communicate my own understanding, so do appreciate the read of this :)

Reply:
I think the new wording works :+1: